web-wallet/src/lib/components/Allocate/Allocate.svelte

```diff
@@ -0,0 +1,417 @@
+  export let execute;
+
+  /** @type {(amount: number) => string} */
+  export let formatter;
+
+  /** @type {ContractGasSettings} */
+  export let gasSettings;
+
+  /** @type {GasStoreContent} */
+  export let gasLimits;
+
+  /** @type {string} */
+  export let shieldedAddress;
+
+  /** @type {string} */
+  export let unshieldedAddress;
+
+  /** @type {number} */
+  export let shieldedBalance;
+
+  /** @type {number} */
+  export let unshieldedBalance;
+
+  /** @type {boolean} */
+  let isNextButtonDisabled = false;
+
+  /** @type {boolean} */
+  let isGasValid = false;
+
+  let { gasLimit, gasPrice } = gasSettings;
+
+  /** @type {number} */
+  let shieldedAmount = shieldedBalance;
+
+  /** @type {number} */
+  let unshieldedAmount = unshieldedBalance;
+
+  /** @type {number} */
+  let screenWidth = window.innerWidth;
+
+  const minAmount = 0.000000001;
+  const dispatch = createEventDispatcher();
+  const resetOperation = () => dispatch("operationChange", "");
+
+  onMount(() => {
+    isGasValid = areValidGasSettings(gasPrice, gasLimit);
+
+    const resizeObserver = new ResizeObserver((entries) => {
+      const entry = entries[0];
+
+      screenWidth = entry.contentRect.width;
+    });
+
+    resizeObserver.observe(document.body);
+
+    return () => resizeObserver.disconnect();
+  });
+
+  $: luxFee = gasLimit * gasPrice;
+  $: fee = formatter(luxToDusk(luxFee));
+  $: isFromUnshielded = shieldedAmount > shieldedBalance;
+  $: isFromShielded = unshieldedAmount > unshieldedBalance;
+  $: isNextButtonDisabled = !(isFromUnshielded || isFromShielded);
+</script>
+
+<div class="operation">
+  <Wizard steps={3} let:key>
+    <WizardStep
+      step={0}
+      {key}
+      backButton={{
+        disabled: false,
+        href: "/dashboard",
+        isAnchor: true,
+      }}
+      nextButton={{ disabled: isNextButtonDisabled }}
+    >
+      <div in:fade|global class="operation__send">
+        <p>
+          Edit the value to change the allocation of your Dusk between your
+          shielded or public account.
+        </p>
+
+        <fieldset class="operation__fieldset">
+          <div class="operation__send-amount operation__label">Shielded</div>
+
+          <div
+            class="operation__send-amount operation__address operation__space-between"
+          >
+            <Icon path={mdiShieldLock} />
+            {middleEllipsis(
+              shieldedAddress,
+              calculateAdaptiveCharCount(screenWidth, 320, 640, 5, 20)
+            )}
+          </div>
+
+          <div class="operation__send-amount operation__input">
+            <Textbox
+              className="operation__input-field"
+              bind:value={shieldedAmount}
+              required
+              type="number"
+              min={minAmount}
+              max={deductLuxFeeFrom(
+                shieldedBalance + unshieldedBalance,
+                luxFee
+              )}
+              step="0.000000001"
```

Comment:
Just a consideration, not related to this PR: I think we should consider switching to `type="text"` textboxes with validation, instead of having number textboxes with this "unusual" steps.

Reply:
I opened an issue in the backlog so we don't forget about this.